grid_detector/src/GridDetector.cpp

```
@@ -1,7 +1,18 @@
-#define ETHERNET 0
-#define WIFI 1
-#define GRID 1
-#define ARDUINO_OTA_ENABLED 1
+#ifndef WIFI
+#  error "WIFI is not defined — use a PlatformIO environment (grid_detector_wifi or grid_detector_eth)"
+#endif
+#ifndef ETHERNET
+#  error "ETHERNET is not defined — use a PlatformIO environment (grid_detector_wifi or grid_detector_eth)"
+#endif
+#if WIFI && ETHERNET
+#  error "Cannot enable both WIFI and ETHERNET simultaneously"
```

Cppcheck (2.21.0)

[error] `#error` "Cannot enable both WIFI and ETHERNET simultaneously"

(preprocessorErrorDirective)